src/mark.js

```diff
@@ -317,7 +317,20 @@ export function values(channels = [], scales) {
       if (scale !== undefined) {
         scale = scales[scale];
         if (scale !== undefined) {
-          value = Array.from(value, scale);
+          if (scale.isCollapsed) {
+            switch(name) {
+              case "x1":
+              case "y1":
+                value = Array.from(value).fill(min(scale.range())); break;
```

Comment:
This is slightly different behavior than my PR this it will apply the scale’s inset. That might be desirable, though it’s different than the behavior of one-dimensional marks, and it’s also now dependent on the way the scale range is defined (which can be overridden). I assume it was done because the dimensions aren’t passed into this method, but it’s worth thinking about.

Also this requires materializing another array which is slow although not likely to be an issue. 😅